via.go

```diff
@@ -0,0 +1,176 @@
+package kramer
+
+import (
+	"bufio"
+	"context"
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"net"
+	"regexp"
+	"time"
+)
+
+const (
+	viaReboot = "Reboot"
+	viaReset  = "Reset"
+)
+
+// VIA Struct that defines general parameters needed for any VIA
+type Via struct {
+	Address  string
+	Username string
+	Password string
+	Logger   Logger
+}
+
+// These functions fulfill the DSP driver requirements
+// GetVolumeByBlock: opening a connection with the VIAs and then return the volume for the device
+func (v *Via) GetVolumeByBlock(ctx context.Context, block string) (int, error) {
+	return v.GetVolume(ctx)
+}
+
+// SetVolumeByBlock: Connect and set the volume on the VIA
+func (v *Via) SetVolumeByBlock(ctx context.Context, block string, volume int) (string, error) {
+	return v.SetVolume(ctx, volume)
+}
+
+// GetMutedByBlock: Return error because VIAs do not support a mute function
+func (v *Via) GetMutedByBlock(ctx context.Context, block string) (bool, error) {
+	return false, errors.New("Error in getting mute status of VIA, Feature not supported")
+}
+
+// SetMutedByBlock: Return error because VIAs do not support mute
+func (v *Via) SetMutedByBlock(ctx context.Context, block string, muted bool) error {
+	return errors.New("Error setting mute status of VIA, Feature not supported")
+}
+
+// GetInfo: needed by the DSP drivers implementation.  Will get hardware information
+func (v *Via) GetInfo(ctx context.Context) (interface{}, error) {
+	var info interface{}
+	return info, fmt.Errorf("GetInfo has not been implemented in this version of the driver")
+}
+
+// End of DSP Driver requirements
+
+func getConnection(address string) (*net.TCPConn, error) {
+	radder, err := net.ResolveTCPAddr("tcp", address+":9982")
+	if err != nil {
+		err = fmt.Errorf("error resolving address : %s", err.Error())
+		return nil, err
+	}
+
+	conn, err := net.DialTCP("tcp", nil, radder)
+	if err != nil {
+		err = fmt.Errorf("error dialing address : %s", err.Error())
+		return nil, err
+	}
+
+	return conn, nil
+}
+
+// SendCommand opens a connection with <addr> and sends the <command> to the via, returning the response from the via, or an error if one occured.
+func (v *Via) sendCommand(ctx context.Context, cmd command) (string, error) {
+	// get the connection
+	v.Infof("Opening telnet connection with %s", v.Address)
+	conn, err := getConnection(v.Address)
+	if err != nil {
+		return "", err
+	}
+
+	timeoutDuration := 7 * time.Second
+
+	// Set Read Connection Duration
+	conn.SetReadDeadline(time.Now().Add(timeoutDuration))
+
+	// login
+	err = v.login(ctx, conn)
+	if err != nil {
+		v.Debugf("Houston, we have a problem logging in. The login failed")
```

Comment:
Let's add in the error that we got back here for some clarity
```suggestion
		v.Debugf("Houston, we have a problem logging in. The login failed: %s", err)
```